src/engine/qml.js

```diff
@@ -157,7 +157,8 @@ function mergeObjects(obj1, obj2) {
   return mergedObject;
 }
 
-const perContextConstructors = {};
+var importContextIds = 0;
+const perImportContextConstructors = {};
```

Comment:
@ChALkeR This may be a future cleanup, but thoughts on keying this off of the Component filename instead of an `importContextId`? I feel it could make things easier to debug, as it would be pretty clear if a chunk of code inherited the imports of a different file.